core/src/main/java/org/apache/iceberg/avro/AvroCustomOrderSchemaVisitor.java

```diff
@@ -90,6 +96,10 @@ public T map(Schema map, Supplier<T> value) {
     return null;
   }
 
+  public T variant(Schema variant) {
```

Comment:
Missing futures for metadata and value fields, like map and array have element and value suppliers.